easybuild/tools/filetools.py

```diff
@@ -1170,34 +1181,47 @@ def adjust_permissions(name, permissionBits, add=True, onlyfiles=False, onlydirs
             # don't change permissions if path is a symlink, since we're not checking where the symlink points to
             # this is done because of security concerns (symlink may point out of installation directory)
             # (note: os.lchmod is not supported on Linux)
-            if not os.path.islink(path):
-                if relative:
+            if os.path.islink(path):
+                _log.debug("Not changing permissions for %s, since it's a symlink", path)
+            else:
+                # determine current permissions
+                current_perms = os.lstat(path)[stat.ST_MODE]
+                _log.debug("Current permissions for %s: %s", path, oct(current_perms))
 
+                if relative:
                     # relative permissions (add or remove)
-                    perms = os.lstat(path)[stat.ST_MODE]
-
                     if add:
-                        os.chmod(path, perms | permissionBits)
+                        _log.debug("Adding permissions for %s: %s", path, oct(permission_bits))
```

Comment:
I would move this below new_perms and print both the permission_bits and the final result.
Same for removing

Reply:
But then I need another `if add:` to discriminate below adding/removing of the `permission_bits`?

The log messages now mention everything: original permissions, added/removed permissions, changed permissions. For (fictional) example:

```
== 2019-12-12 21:51:39,936 filetools.py:1189 DEBUG Current permissions for /Users/kehoste/software/bzip2/1.0.6/bin/bzip2: 0100777
== 2019-12-12 21:51:39,936 filetools.py:1197 DEBUG Removing permissions for /Users/kehoste/software/bzip2/1.0.6/bin/bzip2: 022
== 2019-12-12 21:51:39,936 filetools.py:1207 DEBUG Changing permissions for /Users/kehoste/software/bzip2/1.0.6/bin/bzip2 to 0100755
```

Reply:
Nah, I meant moving it one line down after the new_perms = ... line.
I didn't think of the log below with the actual chmod